odoo/orm/models.py

```diff
@@ -2111,8 +2113,13 @@ def _read_group_groupby(self, alias: str, groupby_spec: str, query: Query) -> SQ
                 # If the granularity uses date_trunc, we need to convert the timestamp back to a date.
                 sql_expr = SQL("%s::date", sql_expr)
 
-        elif field.type == 'boolean':
-            sql_expr = SQL("COALESCE(%s, FALSE)", sql_expr)
+        if field.falsy_value is not None and field not in self.env.registry.not_null_fields:
+            # Use NULLIF to avoid messing the ORDER,
```

Comment:
I think you should coalesce instead. Example: -1 < 0/null < 1

Reply:
I used the NULLIF for keeping the string ordering. Indeed for the integer, that's better to use the COALESCE.

Or we always use COALESCE and change the empty/null string order?